### ISSUE TITLE
iOS Beacon form prefilled custom fields are visible

### ISSUE THREAD

**Issue by ten-aco** [helpscout/beacon-ios-sdk#58]
Hello,

We are using Beacon Library 1.0.5. We are currently having a problem with prefilled custom fields. FInd the code below.

```
import UIKit
import Beacon

class HelpScoutLauncher: NSObject {
    internal func launchHelpScout<T:HSBeaconDelegate>(withVC: T) {
        guard let currentUSer: UserDetails = UserDetails.readAllAssets()?.first else {return}
        let user = HSBeaconUser()
        user.email = currentUSer.email
        user.name = currentUSer.firstName + " " + currentUSer.lastName
        user.addAttribute(withKey: "SF Account ID", value: currentUSer.salesForceID)
        user.addAttribute(withKey: "Case Origin", value: "TFA iOS")
        user.addAttribute(withKey: "App Version", value: EnGestionFields.appVersionBuild ?? "")
        HSBeacon.login(user)
        let settings = HSBeaconSettings(beaconId: "XXXXXXXXXXX")
///////Removed Beacon ID for security
        settings.delegate = withVC
        let messageSettings = HSBeaconMessagingSettings()
        messageSettings.showPrefilledCustomFields = false
        HSBeacon.open(settings)
    }
}


extension UIViewController: HSBeaconDelegate {
    public func prefill(_ form: HSBeaconContactForm) {
        guard let currentUSer: UserDetails = UserDetails.readAllAssets()?.first else {return}
        form.name = currentUSer.firstName + " " + currentUSer.lastName
        form.email = currentUSer.email
        form.addCustomFieldValue(GlobalProperties.na.rawValue, forId: 15623)
        form.addCustomFieldValue("TFA iOS", forId: 15904)
        form.addCustomFieldValue(EnGestionFields.appVersionBuild ?? "", forId: 15905)

    }
}
```

We have been using this code since beginning, we did not have any problem before. But right now, even if we prefill the form, we still see the custom fields. 

* Beacon install method: [ ] Carthage, [X] CocoaPods, [ ] Manual

**Actual outcome**
Beacon shows the prefilled form fields

**Expected outcome**
Beacon should hide the prefilled form fields based on messageSettings.showPrefilledCustomFields = false setting.

**Screenshots or GIFs**
![IMG_35CB15B0B194-1(1)](https://user-images.githubusercontent.com/47034711/79480946-d0277180-7fdc-11ea-8814-cd620389a0af.jpeg)

**Comment by shaunmcilroy** [comment]
Hi @ten-aco thanks for logging this report. Can I get a confirmation on when you upgraded to v1.0.5, and whether it was only after that you observed that pre-filled Custom Fields were no longer being hidden from view? I want to make sure what when you say it didn't happen before whether you're referring to 1.0.5 or a previous version. Thanks!

**Comment by ten-ras** [comment]
Hi shaunmcilroy, We upgraded to v1.0.5 on March 10th. Before we upgraded to v1.0.5, pre-filled Custom Fields were being hidden. We upgraded from version v1.0.1 to v1.0.5. Thanks

**Comment by shaunmcilroy** [comment]
Hey @ten-ras just a quick note to let you know we haven't forgotten or missed this. I wanted to touch base as we recently rolled out v2.0.0 of the SDK, which includes the ability to offer Live Chat for your users so it's possible this behaviour doesn't manifest in that version. The overall functionality is the same, the beta label applies to our first public milestone for offering limited Live Chat functionality to your customers.

I'll check in with our support engineers today and see if we're able to confirm if this is reproducible on the latest version (though you can also opt to test the latest version too if you like!). We'll keep you in the loop!

**Comment by samsymons** [comment]
@ten-ras @ten-aco Hey there! 👋 Sam from the iOS team here, I looked into this issue and confirmed that this appears to be a bug in the SDK.

I don't have a workaround yet beyond publishing a new version of the SDK with the fix, but if I learn anything more I'll let you know. In the meantime, I'll work to get this tidied up and into the next release of the SDK.

Thanks for the bug report! ♥️

**Comment by ten-aco** [comment]
@samsymons Thanks for the update, do you have an ETA for the next release? At least a time frame would be helpful, currently our customer can not fill them form and submit. We may want to hide custom fields within beacon temporarily.

**Comment by shaunmcilroy** [comment]
Hey @ten-aco we're not able to offer specific timelines for when that'll become available, however I would recommend you follow the best route for your needs until we update the SDK in a future release. We'll update our logs to make sure you're followed up with when that happens.

**Comment by samsymons** [comment]
Internal Jira ticket: https://helpscout.atlassian.net/browse/IOSSDK-152

**Comment by adrianne-helpscout** [comment]
👋 @ten-aco,
The fix for this bug went out in 2.0.0-beta.2; so, if you updated to the latest version you should be good to go. Please let us know if you run into any other issues.
